### PR TITLE
GEODE-2484: Replace ACE Map with synchronized unordered_map

### PR DESCRIPTION
**Pull request by pivotal-jbarrett**
* Adds an synchronized_map wrapper template to synchronize a map with a
  mutex.

* Replace unsynchronized ACE Map with std::unordered_map.

### CODE REVIEW

File: cppcache/src/ThinClientRedundancyManager.hpp [hunk 1]
```diff
@@ -26,6 +26,8 @@
 #include <set>
 #include <string>
 
+#include <ace/ACE.h>
```
Comment (pdxcodemonkey) [inline cppcache/src/ThinClientRedundancyManager.hpp]: NNnnnoooooooooooo!  Actually think this is vestigial from some earlier change you made or something.  We removed it and built successfully.

File: cppcache/integration-test/testXmlCacheCreationWithOverFlow.cpp [hunk 2]
```diff
@@ -116,7 +116,7 @@ int testXmlCacheCreationWithOverflow() {
   }
 
   std::cout << "Test if the nesting of regions is correct" << std::endl;
-  auto regPtr2 = vrp.at(1);
+  auto regPtr2 = cptr->getRegion("Root2");
```
[the remark this answers is not in the document]
Comment (pivotal-jbarrett) [reply cppcache/integration-test/testXmlCacheCreationWithOverFlow.cpp]: I’m not changing all global search and replaced lines bad variable names. They will happen if I’m actually manually editing something.

File: cppcache/src/CacheImpl.cpp [hunk 3]
```diff
@@ -467,14 +454,14 @@ void CacheImpl::createRegion(std::string name,
...
-            LOGFINE("enqueued region " + name +
-                    " for initial metadata refresh for singlehop ");
-            poolDM->getClientMetaDataService()->enqueueForMetadataRefresh(
-                regionPtr->getFullPath(), 0);
+          if (auto poolDM = std::dynamic_pointer_cast<ThinClientPoolDM>(pool)) {
+            if (auto clientMetaDataService =
+                    poolDM->getClientMetaDataService()) {
+              LOGFINE("enqueued region " + name +
+                      " for initial metadata refresh for singlehop ");
+              poolDM->getClientMetaDataService()->enqueueForMetadataRefresh(
```
[the remark this answers is not in the document]
Comment (pivotal-jbarrett) [reply cppcache/src/CacheImpl.cpp]: Geez guys, focus on the lines I actually change not just lines that get formatting changes! Good find though.

**Comment by pdxcodemonkey** [comment]
I'm happy to approve this PR after you let me know what you want to do about addressing the namespace issues pointed out in SerializationRegistry.cpp.  Thanks!

**Comment by pivotal-jbarrett** [comment]
@pdxcodemonkey Ready for final review.

**Pull request merged**